// cm6-theme-solarized-dark/dist/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var view = require('@codemirror/view');
var language = require('@codemirror/language');
var highlight = require('@lezer/highlight');

const base00 = '#002b36', base01 = '#073642', base02 = '#586e75', base03 = '#657b83', base04 = '#839496', base05 = '#93a1a1', base06 = '#eee8d5', base07 = '#fdf6e3', base_red = '#dc322f', base_orange = '#cb4b16', base_yellow = '#b58900', base_green = '#859900', base_cyan = '#2aa198', base_blue = '#268bd2', base_violet = '#6c71c4', base_magenta = '#d33682';
const invalid = '#d30102', stone = base04, darkBackground = '#00252f', highlightBackground = '#173541', background = base00, tooltipBackground = base01, selection = '#173541', cursor = base04;
/**
The editor theme styles for Solarized Dark.
*/
const solarizedDarkTheme = view.EditorView.theme({
    '&': {
        color: base05,
        backgroundColor: background
    },
    '.cm-content': {
        caretColor: cursor
    },
    '.cm-cursor, .cm-dropCursor': { borderLeftColor: cursor },
    '&.cm-focused .cm-selectionBackground, .cm-selectionBackground, .cm-content ::selection': { backgroundColor: selection },
    '.cm-panels': { backgroundColor: darkBackground, color: base03 },
    '.cm-panels.cm-panels-top': { borderBottom: '2px solid black' },
    '.cm-panels.cm-panels-bottom': { borderTop: '2px solid black' },
    '.cm-searchMatch': {
        backgroundColor: '#72a1ff59',
        outline: '1px solid #457dff'
    },
    '.cm-searchMatch.cm-searchMatch-selected': {
        backgroundColor: '#6199ff2f'
    },
    '.cm-activeLine': { backgroundColor: highlightBackground },
    '.cm-selectionMatch': { backgroundColor: '#aafe661a' },
    '&.cm-focused .cm-matchingBracket, &.cm-focused .cm-nonmatchingBracket': {
        outline: `1px solid ${base06}`
    },
    '.cm-gutters': {
        backgroundColor: darkBackground,
        color: stone,
        border: 'none'
    },
    '.cm-activeLineGutter': {
        backgroundColor: highlightBackground
    },
    '.cm-foldPlaceholder': {
        backgroundColor: 'transparent',
        border: 'none',
        color: '#ddd'
    },
    '.cm-tooltip': {
        border: 'none',
        backgroundColor: tooltipBackground
    },
    '.cm-tooltip .cm-tooltip-arrow:before': {
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent'
    },
    '.cm-tooltip .cm-tooltip-arrow:after': {
        borderTopColor: tooltipBackground,
        borderBottomColor: tooltipBackground
    },
    '.cm-tooltip-autocomplete': {
        '& > ul > li[aria-selected]': {
            backgroundColor: highlightBackground,
            color: base03
        }
    }
}, { dark: true });
/**
The highlighting style for code in the Solarized Dark theme.
*/
const solarizedDarkHighlightStyle = language.HighlightStyle.define([
    { tag: highlight.tags.keyword, color: base_green },
    {
        tag: [highlight.tags.name, highlight.tags.deleted, highlight.tags.character, highlight.tags.propertyName, highlight.tags.macroName],
        color: base_cyan
    },
    { tag: [highlight.tags.variableName], color: base05 },
    { tag: [highlight.tags.function(highlight.tags.variableName)], color: base_blue },
    { tag: [highlight.tags.labelName], color: base_magenta },
    {
        tag: [highlight.tags.color, highlight.tags.constant(highlight.tags.name), highlight.tags.standard(highlight.tags.name)],
        color: base_yellow
    },
    { tag: [highlight.tags.definition(highlight.tags.name), highlight.tags.separator], color: base_cyan },
    { tag: [highlight.tags.brace], color: base_magenta },
    {
        tag: [highlight.tags.annotation],
        color: invalid
    },
    {
        tag: [highlight.tags.number, highlight.tags.changed, highlight.tags.annotation, highlight.tags.modifier, highlight.tags.self, highlight.tags.namespace],
        color: base_magenta
    },
    {
        tag: [highlight.tags.typeName, highlight.tags.className],
        color: base_orange
    },
    {
        tag: [highlight.tags.operator, highlight.tags.operatorKeyword],
        color: base_violet
    },
    {
        tag: [highlight.tags.tagName],
        color: base_blue
    },
    {
        tag: [highlight.tags.squareBracket],
        color: base_red
    },
    {
        tag: [highlight.tags.angleBracket],
        color: base02
    },
    {
        tag: [highlight.tags.attributeName],
        color: base05
    },
    {
        tag: [highlight.tags.regexp],
        color: invalid
    },
    {
        tag: [highlight.tags.quote],
        color: base_green
    },
    { tag: [highlight.tags.string], color: base_yellow },
    {
        tag: highlight.tags.link,
        color: base_cyan,
        textDecoration: 'underline',
        textUnderlinePosition: 'under'
    },
    {
        tag: [highlight.tags.url, highlight.tags.escape, highlight.tags.special(highlight.tags.string)],
        color: base_yellow
    },
    { tag: [highlight.tags.meta], color: base_red },
    { tag: [highlight.tags.comment], color: base02, fontStyle: 'italic' },
    { tag: highlight.tags.strong, fontWeight: 'bold', color: base06 },
    { tag: highlight.tags.emphasis, fontStyle: 'italic', color: base_green },
    { tag: highlight.tags.strikethrough, textDecoration: 'line-through' },
    { tag: highlight.tags.heading, fontWeight: 'bold', color: base_yellow },
    { tag: highlight.tags.heading1, fontWeight: 'bold', color: base07 },
    {
        tag: [highlight.tags.heading2, highlight.tags.heading3, highlight.tags.heading4],
        fontWeight: 'bold',
        color: base06
    },
    {
        tag: [highlight.tags.heading5, highlight.tags.heading6],
        color: base06
    },
    { tag: [highlight.tags.atom, highlight.tags.bool, highlight.tags.special(highlight.tags.variableName)], color: base_magenta },
    {
        tag: [highlight.tags.processingInstruction, highlight.tags.inserted, highlight.tags.contentSeparator],
        color: base_red
    },
    {
        tag: [highlight.tags.contentSeparator],
        color: base_yellow
    },
    { tag: highlight.tags.invalid, color: base02, borderBottom: `1px dotted ${base_red}` }
]);
/**
Extension to enable the Solarized Dark theme (both the editor theme and
the highlight style).
*/
const solarizedDark = [
    solarizedDarkTheme,
    language.syntaxHighlighting(solarizedDarkHighlightStyle)
];

exports.solarizedDark = solarizedDark;
exports.solarizedDarkHighlightStyle = solarizedDarkHighlightStyle;
exports.solarizedDarkTheme = solarizedDarkTheme;
