// cm6-theme-basic-dark/dist/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var view = require('@codemirror/view');
var language = require('@codemirror/language');
var highlight = require('@lezer/highlight');

const base00 = '#2E3235', base01 = '#DDDDDD', base02 = '#B9D2FF', base03 = '#b0b0b0', base05 = '#e0e0e0', base06 = '#808080', base07 = '#000000', base08 = '#A54543', base09 = '#fc6d24', base0A = '#fda331', base0B = '#8abeb7', base0C = '#b5bd68', base0D = '#6fb3d2', base0E = '#cc99cc', base0F = '#6987AF';
const invalid = base09, darkBackground = '#292d30', highlightBackground = base02 + '30', background = base00, tooltipBackground = base01, selection = '#202325', cursor = base01;
/**
The editor theme styles for Basic Dark.
*/
const basicDarkTheme = view.EditorView.theme({
    '&': {
        color: base01,
        backgroundColor: background
    },
    '.cm-content': {
        caretColor: cursor
    },
    '.cm-cursor, .cm-dropCursor': { borderLeftColor: cursor },
    '&.cm-focused .cm-selectionBackground, .cm-selectionBackground, .cm-content ::selection': { backgroundColor: selection },
    '.cm-panels': { backgroundColor: darkBackground, color: base03 },
    '.cm-panels.cm-panels-top': { borderBottom: '2px solid black' },
    '.cm-panels.cm-panels-bottom': { borderTop: '2px solid black' },
    '.cm-searchMatch': {
        backgroundColor: base02,
        outline: `1px solid ${base03}`,
        color: base07
    },
    '.cm-searchMatch.cm-searchMatch-selected': {
        backgroundColor: base05,
        color: base07
    },
    '.cm-activeLine': { backgroundColor: highlightBackground },
    '.cm-selectionMatch': { backgroundColor: highlightBackground },
    '&.cm-focused .cm-matchingBracket, &.cm-focused .cm-nonmatchingBracket': {
        outline: `1px solid ${base03}`
    },
    '&.cm-focused .cm-matchingBracket': {
        backgroundColor: base02,
        color: base07
    },
    '.cm-gutters': {
        borderRight: `1px solid #ffffff10`,
        color: base06,
        backgroundColor: darkBackground
    },
    '.cm-activeLineGutter': {
        backgroundColor: highlightBackground
    },
    '.cm-foldPlaceholder': {
        backgroundColor: 'transparent',
        border: 'none',
        color: base02
    },
    '.cm-tooltip': {
        border: 'none',
        backgroundColor: tooltipBackground
    },
    '.cm-tooltip .cm-tooltip-arrow:before': {
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent'
    },
    '.cm-tooltip .cm-tooltip-arrow:after': {
        borderTopColor: tooltipBackground,
        borderBottomColor: tooltipBackground
    },
    '.cm-tooltip-autocomplete': {
        '& > ul > li[aria-selected]': {
            backgroundColor: highlightBackground,
            color: base03
        }
    }
}, { dark: true });
/**
The highlighting style for code in the Basic Light theme.
*/
const basicDarkHighlightStyle = language.HighlightStyle.define([
    { tag: highlight.tags.keyword, color: base0A },
    {
        tag: [highlight.tags.name, highlight.tags.deleted, highlight.tags.character, highlight.tags.propertyName, highlight.tags.macroName],
        color: base0C
    },
    { tag: [highlight.tags.variableName], color: base0D },
    { tag: [highlight.tags.function(highlight.tags.variableName)], color: base0A },
    { tag: [highlight.tags.labelName], color: base09 },
    {
        tag: [highlight.tags.color, highlight.tags.constant(highlight.tags.name), highlight.tags.standard(highlight.tags.name)],
        color: base0A
    },
    { tag: [highlight.tags.definition(highlight.tags.name), highlight.tags.separator], color: base0E },
    { tag: [highlight.tags.brace], color: base0E },
    {
        tag: [highlight.tags.annotation],
        color: invalid
    },
    {
        tag: [highlight.tags.number, highlight.tags.changed, highlight.tags.annotation, highlight.tags.modifier, highlight.tags.self, highlight.tags.namespace],
        color: base0A
    },
    {
        tag: [highlight.tags.typeName, highlight.tags.className],
        color: base0D
    },
    {
        tag: [highlight.tags.operator, highlight.tags.operatorKeyword],
        color: base0E
    },
    {
        tag: [highlight.tags.tagName],
        color: base0A
    },
    {
        tag: [highlight.tags.squareBracket],
        color: base0E
    },
    {
        tag: [highlight.tags.angleBracket],
        color: base0E
    },
    {
        tag: [highlight.tags.attributeName],
        color: base0D
    },
    {
        tag: [highlight.tags.regexp],
        color: base0A
    },
    {
        tag: [highlight.tags.quote],
        color: base01
    },
    { tag: [highlight.tags.string], color: base0C },
    {
        tag: highlight.tags.link,
        color: base0F,
        textDecoration: 'underline',
        textUnderlinePosition: 'under'
    },
    {
        tag: [highlight.tags.url, highlight.tags.escape, highlight.tags.special(highlight.tags.string)],
        color: base0B
    },
    { tag: [highlight.tags.meta], color: base08 },
    { tag: [highlight.tags.comment], color: base06, fontStyle: 'italic' },
    { tag: highlight.tags.monospace, color: base01 },
    { tag: highlight.tags.strong, fontWeight: 'bold', color: base0A },
    { tag: highlight.tags.emphasis, fontStyle: 'italic', color: base0D },
    { tag: highlight.tags.strikethrough, textDecoration: 'line-through' },
    { tag: highlight.tags.heading, fontWeight: 'bold', color: base01 },
    { tag: highlight.tags.special(highlight.tags.heading1), fontWeight: 'bold', color: base01 },
    { tag: highlight.tags.heading1, fontWeight: 'bold', color: base01 },
    {
        tag: [highlight.tags.heading2, highlight.tags.heading3, highlight.tags.heading4],
        fontWeight: 'bold',
        color: base01
    },
    {
        tag: [highlight.tags.heading5, highlight.tags.heading6],
        color: base01
    },
    { tag: [highlight.tags.atom, highlight.tags.bool, highlight.tags.special(highlight.tags.variableName)], color: base0B },
    {
        tag: [highlight.tags.processingInstruction, highlight.tags.inserted],
        color: base0B
    },
    {
        tag: [highlight.tags.contentSeparator],
        color: base0D
    },
    { tag: highlight.tags.invalid, color: base02, borderBottom: `1px dotted ${invalid}` }
]);
/**
Extension to enable the Basic Dark theme (both the editor theme and
the highlight style).
*/
const basicDark = [
    basicDarkTheme,
    language.syntaxHighlighting(basicDarkHighlightStyle)
];

exports.basicDark = basicDark;
exports.basicDarkHighlightStyle = basicDarkHighlightStyle;
exports.basicDarkTheme = basicDarkTheme;
